engine/sources/gitdns/source.go

```diff
@@ -247,9 +255,23 @@ func (gs *gitSourceHandler) getAuthToken(ctx context.Context, g session.Group) e
 				}
 				return err
 			}
+
+			// Handle token authentication
 			if s.token {
+				if s.credHelper {
```

Comment:
nit: is there a way to get this from the client config instead?

as @vito mentioned above, we might have a problem with GH enterprise as well